zcash_primitives/src/transaction/tests.rs

```diff
@@ -313,7 +314,14 @@ fn zip_0244() {
         (tdata, txdata.digest(TxIdDigester))
     }
 
-    for tv in self::data::zip_0244::make_test_vectors() {
+    #[cfg(not(zcash_unstable = "nu6" /* TODO nu7 */ ))]
+    let test_vectors = self::data::zip_0244::make_test_vectors();
+    #[cfg(zcash_unstable = "nu6" /* TODO nu7 */ )]
+    let mut test_vectors = self::data::zip_0244::make_test_vectors();
```

Comment:
Are we doing all this just to avoid 'mut' in case of not nu6? Seems like too much to me

Reply:
Yeah, should I use `#[allow(unused_mut)]` instead?

Reply:
Done.